tests/macros/defaultapisidebar.test.js

```diff
@@ -0,0 +1,359 @@
+        // Check the list
+        const actualItems = actual.querySelectorAll('ol>li');
+        const expectedItems = config.expected.details[name];
+        expect(actualItems.length).toEqual(expectedItems.length);
+        for (let i = 0; i < actualItems.length; i++) {
+            checker(expectedItems[i], actualItems[i], config.locale);
+        }
+        next++;
+    }
+    return next;
+}
+
+/**
+* This is the entry point for checking the result of a test.
+* config.expected contains the expected results, and we use other bits
+* of config, most notably locale.
+*/
+function checkResult(html, config) {
+    // Lint the HTML
+    expect(lintHTML(html)).toBeFalsy();
+
+    const dom = JSDOM.fragment(html);
+    // Check that all links reference the proper locale or use https
+    const num_total_links = dom.querySelectorAll('a[href]').length;
+    const num_valid_links = dom.querySelectorAll(`a[href^="/${config.locale}/"], a[href^="https://"]`).length;
+    expect(num_valid_links).toEqual(num_total_links);
+
+    if (config.expected.overview) {
+        // Test overview link
+        const overviewLink = dom.querySelector('ol>li>strong');
+        checkItem(config.expected.overview, overviewLink, config.locale);
+    }
+
+    // Test sublists
+    const details = dom.querySelectorAll('ol>li>details');
+    expect(details.length).toEqual(Object.keys(config.expected.details).length);
+    // "_" prefix is to suppress lint's no-unused-vars
+    let _next = 0;
+    _next = checkSubList('Guides', config, details, checkGuideItem, _next);
+    _next = checkSubList('Interfaces', config, details, checkItem, _next);
+    _next = checkSubList('Properties', config, details, checkItem, _next);
+    _next = checkSubList('Methods', config, details, checkItem, _next);
+    _next = checkSubList('Events', config, details, checkItem, _next);
+}
+
+/**
+* Call the macro for each of three locales,
+* using the given config, and check the result.
+*/
+function testMacro(config) {
+    for (const locale of ['en-US', 'fr', 'ja']) {
+        let testName = `${config.name}; locale: ${locale}`;
+        itMacro(testName, function(macro) {
+            config.locale = locale;
+            macro.ctx.env.locale = locale;
+            // Mock calls to MDN.subpagesExpand
+            macro.ctx.page.subpagesExpand = jest.fn((page) => {
+                return config.subpages;
+            });
+            return macro.call(config.argument).then(function(result) {
+                checkResult(result, config);
+            });
+
+        });
+    }
+}
+
+describeMacro('DefaultAPISidebar', function() {
+
+    // Set any fixtures that don't change from one
+    // test to another
+    beforeEachMacro(function(macro) {
+        // env.slug only has to be truthy
+        macro.ctx.env.slug = 'not undefined';
+        // Mock calls to L10n-Common and GroupData
+        const originalTemplate = macro.ctx.template;
+        macro.ctx.template = jest.fn( async (name, ...args) => {
```

Comment:
I’d prefer if this used `macro.mockTemplate(…)` from https://github.com/mdn/kumascript/pull/1093.